<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1068</width>
    <height>604</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>591</width>
      <height>341</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="vertsListWidget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>10</y>
      <width>61</width>
      <height>321</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="halfEdgesListWidget">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>10</y>
      <width>61</width>
      <height>321</height>
     </rect>
    </property>
   </widget>
   <widget class="QListWidget" name="facesListWidget">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>10</y>
      <width>61</width>
      <height>321</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>605</x>
      <y>340</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertices</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>685</x>
      <y>340</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Half-Edges</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>765</x>
      <y>340</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Faces</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>766</x>
      <y>391</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
    <property name="maximum">
     <double>99.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>883</x>
      <y>391</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="vertPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>993</x>
      <y>391</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>617</x>
      <y>392</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Position</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>725</x>
      <y>394</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>842</x>
      <y>394</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>953</x>
      <y>394</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>943</x>
      <y>424</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Blue</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>718</x>
      <y>424</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Red</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceBlueSpinBox">
    <property name="geometry">
     <rect>
      <x>992</x>
      <y>421</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>617</x>
      <y>424</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Face Color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceGreenSpinBox">
    <property name="geometry">
     <rect>
      <x>882</x>
      <y>421</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="faceRedSpinBox">
    <property name="geometry">
     <rect>
      <x>766</x>
      <y>421</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>1.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>827</x>
      <y>424</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Green</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="addVertex">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>422</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Add Vertex</string>
    </property>
   </widget>
   <widget class="QPushButton" name="triangulate">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>422</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Triangulate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivide">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>453</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QPushButton" name="extrudeFace">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>453</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Extrude Face</string>
    </property>
   </widget>
   <widget class="QPushButton" name="importobj">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Import OBJ </string>
    </property>
   </widget>
   <widget class="QCheckBox" name="sharp">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>360</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Sharp</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>905</x>
      <y>340</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Joints</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="importjson">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>390</y>
      <width>141</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Import JSON </string>
    </property>
   </widget>
   <widget class="QPushButton" name="skinMesh">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>390</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Skin Mesh</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="jointsListWidget">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>10</y>
      <width>181</width>
      <height>321</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">1</string>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>617</x>
      <y>454</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Position (local)</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosYSpinBox">
    <property name="geometry">
     <rect>
      <x>882</x>
      <y>451</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>952</x>
      <y>454</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>841</x>
      <y>454</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>724</x>
      <y>454</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosXSpinBox">
    <property name="geometry">
     <rect>
      <x>765</x>
      <y>451</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
    <property name="maximum">
     <double>99.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="jointPosZSpinBox">
    <property name="geometry">
     <rect>
      <x>992</x>
      <y>451</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-99.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>617</x>
      <y>486</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Joint Rotation</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotX">
    <property name="geometry">
     <rect>
      <x>747</x>
      <y>482</y>
      <width>80</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>rotate X</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotY">
    <property name="geometry">
     <rect>
      <x>864</x>
      <y>482</y>
      <width>80</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>rotate Y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jointRotZ">
    <property name="geometry">
     <rect>
      <x>973</x>
      <y>482</y>
      <width>80</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>rotate Z</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1068</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionCamera_Controls"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionCamera_Controls">
   <property name="text">
    <string>Camera Controls</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
